pkg/serial/handshake.go

```diff
@@ -77,212 +72,188 @@ func HandshakeClient(ctx context.Context, conn net.Conn, debug bool) error {
 		conn.SetReadDeadline(deadline)
 	}
 
-	rand.Read(syn[1:])
-
-	conn.Write(syn)
+	log.Debugf("HandshakeClient: Establishing connection...")
+	for {
```

Comment:
The client should not be running in a for loop at this point - there needs to be a purge incoming call before each iteration to deal with the case of queued incoming bytes - otherwise each byte will trigger sending a new syn packet, which can trigger more NACKs from the server side.

This loop needs to run slower than the consumption on server side to prevent a self-sustaining bouncing of syn/Nack and there's no throttle.